.github/workflows/ci.yaml

```diff
@@ -51,6 +51,17 @@ jobs:
       - name: Run tests
         run: tox -e unit
 
+  type-check:
+    name: Mypy tests
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install dependencies
+        run: python3 -m pip install tox
+      - name: Run tests
+        run: tox -e type
```

Comment:
Are we sure we want to block CI on type checking? 

To be honest, I've always been a bit conflicted about strict type checking in Python since it is a feature that is bolted onto the language rather than being directly built in like Rust or Go. I think it can be helpful for linting, but I don't think it should be a hard requirement. Reason saying is that it's easy to become lazy and add `#type: ignore` everywhere where we don't want to spend typing. It's also not necessarily enforced at runtime either. I do think type hinting as documentation is good for expected types.

Let me know you thoughts.

Reply:
I do like the type-checking. Omnivector goes to great lengths in other code projects to make sure the code is typed, and I have grown to like reading the code with type hints. I wouldn't say I like ignoring the type checks inline, but I was stuck and wanted to move on, so I ignored cases I couldn't solve. I should do a better job of notating my reasoning for neglecting.

If you are opposed to it, possibly we can have a conversation about it in our next community meting?

Reply:
I'm not opposed to it's inclusion, in fact, I think it's good for development because it commits us to accurately documenting methods and what types they accept v.s. don't. Biggest pain is when you have a super nested method that fails because it received a type that you didn't expect it too, or you didn't realize that your object was casted into a different type by another method.

I'm fine with adding it as long as we commit to documenting why `# type: ignore` is used in certain places. Doesn't need to be super extensive, and can be a short note in the commit message. What do you think of that?

Reply:
Love it! Ill add some documentation to my ignores. To wrap up this pr.